ouroboros-network-framework/sim-tests/Test/Ouroboros/Network/Server2/Sim.hs

```diff
@@ -599,13 +598,14 @@ instance Show addr => Show (Name addr) where
     show  MainServer   = "main-server"
 
 
-data ExperimentError addr =
-      NodeNotRunningException addr
-    | NoActiveConnection addr addr
+data ExperimentError =
+      forall addr. (Typeable addr, Show addr) => NodeNotRunningException addr
+    -- | forall addr. (Typeable addr, Show addr) => NoActiveConnection addr addr
```

Comment:
Is this needed?

Reply:
I left it, so we can come back to it at some point.